### PR TITLE
Mobile 4081

### PR DESCRIPTION
**Pull request by alfonso-salces**
This pull request has the following two changes:

- Add possibility of add toggle inside site preferences page.

- Add event to use it when main home loads.

### CODE REVIEW

File: src/core/features/settings/pages/site/site.html [hunk 1]
```diff
@@ -18,13 +18,16 @@ <h1>{{ 'core.settings.preferences' | translate}}</h1>
...
-                    [attr.aria-label]="handler.title | translate" detail="true" (click)="handlers.select(handler)" button
-                    [attr.aria-current]="handlers.getItemAriaCurrent(handler)">
+                    [attr.aria-label]="handler.title | translate" (click)="handlers.select(handler)" [button]="!handler.toggle"
+                    [detail]="!handler.toggle" [attr.aria-current]="handlers.getItemAriaCurrent(handler)">
                     <ion-icon [name]="handler.icon" slot="start" *ngIf="handler.icon" aria-hidden="true">
                     </ion-icon>
                     <ion-label>
                         <p class="item-heading">{{ handler.title | translate}}</p>
                     </ion-label>
+                    <ion-toggle *ngIf="handler.toggleChecked !== undefined" [(ngModel)]="handler.toggleChecked"
```
Comment (NoelDeMartin) [inline src/core/features/settings/pages/site/site.html:28]:
```suggestion
                    <ion-toggle *ngIf="handler.toggle" [(ngModel)]="handler.toggleChecked"
```

This should be more readable and we're already using the same logic on the lines above.

File: src/core/features/settings/services/settings-delegate.ts [hunk 2]
```diff
@@ -52,6 +52,16 @@ export interface CoreSettingsHandlerData {
...
+
+    /**
+     * Toggle checked.
+     */
+    toggleChecked?: boolean;
+
+    /**
+     * Method for emit events to the handler.
+     */
+    toggle?(checked: boolean): void;
```
Comment (NoelDeMartin) [inline src/core/features/settings/services/settings-delegate.ts:64]: I suppose for a toggle handler, both `toggle` and `toggleChecked` are required, right? Also, `page` is probably not needed, but according to the current implementation it'll be required. I'm thinking we could define the specific types, and type guards should take care of inferring what's optional or not.

```ts
interface CoreSettingsHandlerBaseData {
    title: string;
    class?: string;
    icon?: string;
}

interface CoreSettingsToggleHandlerData extends CoreSettingsHandlerBaseData {
    toggleChecked: boolean;
    toggle(checked: boolean): void;
}

interface CoreSettingsPageHandlerData extends CoreSettingsHandlerBaseData {
    page: string;
    params?: Params;
}

type CoreSettingsHandlerData = CoreSettingsPageHandlerData | CoreSettingsToggleHandlerData;
```

**Review by NoelDeMartin — COMMENTED** [review]
I think this is ready for merging, but I have some suggestions, let me know what you think.

File: src/core/features/settings/services/settings-delegate.ts (hunk 2 above)
Comment (alfonso-salces) [reply src/core/features/settings/services/settings-delegate.ts:64]: I had to tried to implement something like your suggestion and it's works but I have to do some narrowing in settings-handlers-source.ts and in site page (site.ts). Maybe that can help to clear settings handlers of unused properties, but I don't just like narrowing.
I have to say, that maybe Im missing other way of implementation for that which don't involves narrowing.

**Review by NoelDeMartin — APPROVED** [review]

**Pull request merged**